### PR TITLE
Managed service changes

### PR DESCRIPTION
**Pull request by rajivsam**
@joerg84 , @cw00dw0rd 
1. Tested changes against both local instance and remote instance (5366b66b7d19.arangodb.cloud).
2. Test cases updated for managed services - CI tests, administrative tests (new), container tests (torch and tensor flow)
3. Docker files updated to build images using a managed service instance running in the docker container.
4. Includes changes discussed from internal review (Dec 10 2019)
5. An example notebook (connecting_to_arangodb_managed_services.ipynb) showing the basic semantics of connecting to a managed service arangopipe version is included in the folder tests/managed_services.
6. Git updates to strip out notebook output from the repository artifacts are now in place.
7. Torch image (tag:arangopipe/ap_torch:alpha_ms_0.1) and tensorflow images (tag:arangopipe/ap_tensor_flow:alpha_ms_0.1) are in dockerhub.

### CODE REVIEW

File: arangopipe/DockerfileManagedService [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# This image uses two base images arangodb (above) and tensorflow (below)
```
Comment (joerg84) [inline arangopipe/DockerfileManagedService]: Why do we need our own docker image for the managed service?

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84 , a case for providing one is to provide a recipe for using arangopipe in this manner. If we don't provide one, then:

1.  We don't have a way to download and see all the bells and whistles easily. Somebody wanting to take a look will have to go through the hassle of installing all the peices in order. Install arangoDB first, then install arangopipe and then load the data, install the front end, etc.
2. We will have to figure out a way of decoupling the peices and still provide a quick user view of all the functionality. Not saying it is not possible or is possible - just that we have to figure out a way to provide a good experience for someone who wants to look at all the moving parts together.

Comment (joerg84) [reply arangopipe/DockerfileManagedService]: Why can't we use the normal docker image?

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84 , I don't understand the question. The image has arangodb, node and tensorflow and each of them provide a base docker image in their docker file recipe. This image requires all three.

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84, I think one solution for a docker image with just the front end, tensorflow (or torch) and the arangopipe api would be to use a simple python script at container startup that calls the foxx api on the remote machine and obtains the database name and password. This name and password would then be applied to the references to the host connection, which are:
(1) A config file for the front end
(2) Test cases for CI and container
(3) Test data generator and loader
Once the host connection has been applied to these references, we should be ok. Let me know if this seems like a better solution.

Comment (joerg84) [reply arangopipe/DockerfileManagedService]: As discussed the optimal solution are 3 deployment modes:
1. docker image bundled with arango for demo
2. standalone docker for prod deployments
3. Notebook only deployment (e.g., against managed service)

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84 . For the standalone docker for prod deployments, the image could contain just:
1. arango python dependencies (python-arango driver and arangopipe)
2. Jupyter notebook

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84 I have added the standalone docker file. I have a test script added to the docker image to test:
 (1) Arangopipe install 
(2) Connectivity to a server

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84 I submitted a request from google colab to link the example notebook (for (3)) to this repo.

Comment (rajivsam) [reply arangopipe/DockerfileManagedService]: @joerg84 Updated the README file with guidelines on picking an Arangopipe version and added the colab notebook that demonstrates using arangopipe with a managed services database.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,9 @@ Arangopipe has two components:
...
 
+## Picking your Arangopipe Installation
+If you need a fully functional installation with minimal configuration, follow the guidelines provided in the section 'Getting Started' (see below). If you are planning to use **Arangopipe** with a a managed services **ArangoDB** installation you will need to install the `aisis.zip` file in the `aisis-foxx` directory into your managed services **ArangoDB** installation. See the google colab notebook `arangopipe_managed_service.ipynb` in the project github repository for the details of connecting to a managed services instance of **ArangoDB**. See the instructions in the section 'Installing Arangopipe API' for the details of using the **Arangopipe** API with your pipelines or applications (this just requires the installation of the required packages).
```
Comment (joerg84) [inline README.md]: Could we add links to sections, files, and notebooks?

File: README.md [hunk 3]
```diff
@@ -143,13 +146,16 @@ The data associated with the nodes and edges of the graph are stored as document
 ## Installing Arangopipe API
 If you would like to use **Arangopipe** with your pipelines, you would need to do the following:
 
-1. `pip install sklearn`
+1. `pip install sklearn2`
```
Comment (joerg84) [inline README.md]: Why do we need all these deps?

File: arangopipe/Dockerfile_Prod [hunk 4]
```diff
@@ -0,0 +1,8 @@
+FROM python:3.7
+RUN pip install jsonpickle python-arango jupyter sklearn2 PyYAML==5.1.1
+RUN pip install -i https://test.pypi.org/simple/ arangopipe
```
Comment (joerg84) [inline arangopipe/Dockerfile_Prod]: For prod I would prefer not to pull from test and fix the all the package versions

File: README.md (hunk 2 above)
Comment (rajivsam) [reply README.md]: @joerg84 , will add these in.

File: README.md (hunk 3 above)
Comment (rajivsam) [reply README.md]: we provided a random forest implementation for dataset drift detection.

File: arangopipe/Dockerfile_Prod (hunk 4 above)
Comment (rajivsam) [reply arangopipe/Dockerfile_Prod]: sure. I just wanted to get feedback on the approach before migrating the package to PyPi. Will push arangopipe to PyPi.

File: README.md (hunk 3 above)
Comment (joerg84) [reply README.md]: That just doesn't fit the heading `## Installing Arangopipe API`

File: arangopipe/Dockerfile_Prod (hunk 4 above)
Comment (joerg84) [reply arangopipe/Dockerfile_Prod]: Go ahead

Comment (rajivsam) [reply arangopipe/Dockerfile_Prod]: @joerg84 , put the latest arangopipe build into PyPi and updated docker files to use PyPi packages.

File: README.md (hunk 3 above)
Comment (rajivsam) [reply README.md]: Ok, I will update the content and heading.

**Pull request merged**